### ISSUE TITLE
Docs: Write blog post on the history of ban-types, {}, and object/Object 

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#9103]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

One of the particularly exciting changes in our upcoming `v8` version is us finally fixing up messaging around the `{}` (empty object type). It's long been a contentious part of the `ban-types` rule:

* On the one hand, `{}` is a confusing type that a lot of TS devs get tripped up over
* On the other hand, there are legitimate use cases for it, and banning it outright is too string

We think we've reached a good compromise for v8 with the combination of:

* For `{}`, splitting out a dedicated `no-empty-object-type` rule: #8700 -> #8977
* For the remaining uses of (a) configurable types and (b) uppercase aliases like `Number`, splitting out a couple of rules: #8978 -> #9102

This has been a long journey with lots of discussion (#8700 is just one of many!). It seems ripe for a blog post to me!

💖 

### Affected URL(s)

https://typescript-eslint.io/blog/*

Note that if this is accepted, I think a member of our team should write the post. It's not something an external contributor could easily do. Blog posts are tricky.

**Comment by Josh-Cena** [comment]
Yes, I think there's a lot of room for education here in the blog post...

**Comment by kirkwaiblinger** [comment]
Linking to https://github.com/microsoft/TypeScript/issues/60582 here for visibility
